hidimstat/estimator/RandomForestModified.py

```diff
@@ -42,23 +42,24 @@ def sample_same_leaf(self, X, y=None):
                 )[0]
 
                 # Append the samples to the list
-                leaf_samples.append(
-                    y_minus_i[rng.choice(samples_in_leaf, size=random_samples)]
-                )
+                if samples_in_leaf.size > 0:
+                    leaf_samples.append(
+                        y_minus_i[rng.choice(samples_in_leaf, size=random_samples)]
+                    )
```

Comment:
@jpaillard @bthirion
I modified the function for considering when the samples leaf was empty.
However, I don't know what was doing the function. 
Can you validate if it's the correct way to do it?

Reply:
The usage of  samples_in_leaf is explain in #42.